refinery/tool_manager/models.py

```diff
@@ -1164,10 +1175,19 @@ def _get_galaxy_workflow_invocation(self):
         """
         Fetch our Galaxy Workflow's invocation data.
         """
-        return self.galaxy_connection.workflows.show_invocation(
-            self.galaxy_workflow_history_id,
-            self.get_galaxy_dict()[self.GALAXY_WORKFLOW_INVOCATION_DATA]["id"]
-        )
+        # separate if-then assignment needed to avoid using the dict stored
+        # in self.invocation before .save() is called
+        if self.invocation == '' or self.invocation is None:
```

Comment:
Comparing to `None` is not necessary if `invocation` cannot be `null` in the database (it shouldn't be).